daffodil-cli/src/it/scala/org/apache/daffodil/udf/TestCLIUdfs.scala

```diff
@@ -0,0 +1,565 @@
+          contains("[warning] User Defined Function ignored:" +
+            " org.badudfs.nonUDF.StringFunctions.Replace." +
+            " Doesn't implement org.apache.daffodil.udf.UserDefinedFunction"),
+          contains("[info] No User Defined Functions loaded."),
+          contains("[error] Schema Definition Error: Unsupported function: jsudf:replace")))
+
+      shell.send("exit\n")
+      shell.expect(eof)
+      shell.close()
+    } finally {
+      shell.close()
+    }
+  }
+
+  /**
+   * Tests the case when a function class doesn't have annotations or have empty/invalid
+   * annotstion fields
+   */
+  @Test def test_UDFClass_nonAnn() {
+    val schemaFile = Util.daffodilPath("daffodil-udf/src/test/resources/org/apache/daffodil/udf/genericUdfSchema.xsd")
+    val (testSchemaFile) = if (Util.isWindows) (Util.cmdConvert(schemaFile)) else (schemaFile)
+
+    val metaInfForSomeUdfA = "daffodil-udf/src/test/java/org/badudfs/annotations/StringFunctions/"
+
+    val dafClassPath =
+      (testUdfsPaths :+ Util.daffodilPath(metaInfForSomeUdfA))
+        .mkString(":")
+
+    val shell = Util.startIncludeErrors("", envp = Map("DAFFODIL_CLASSPATH" -> dafClassPath))
+
+    try {
+      val cmd = String.format("echo -n strng| %s -v parse -s %s -r user_func1", Util.binPath, testSchemaFile)
+      shell.sendLine(cmd)
+      shell.expectIn(
+        1,
+        allOf(
+          contains("[warning] User Defined Function ignored:" +
+            " org.badudfs.annotations.StringFunctions.FuncB." +
+            " Missing org.apache.daffodil.udf.UserDefinedFunctionIdentification annotation"),
+          anyOf(
+            contains("[warning] User Defined Function ignored:" +
+              " org.badudfs.annotations.StringFunctions.Compare." +
+              " Annotation namespace field is empty or invalid."),
+            contains("[warning] User Defined Function ignored:" +
+              " org.badudfs.annotations.StringFunctions.Compare." +
+              " Annotation name field is empty or invalid.")),
+          contains("[warning] User Defined Function ignored:" +
+            " org.badudfs.annotations.StringFunctions.Replace." +
+            " Annotation name field is empty or invalid."),
+          contains("[info] No User Defined Functions loaded."),
+          contains("[error] Schema Definition Error: Unsupported function: jsudf:replace")))
+
+      shell.send("exit\n")
+      shell.expect(eof)
+      shell.close()
+    } finally {
+      shell.close()
+    }
+  }
+
+  /**
+   * Tests the cases when a function class:
+   *   doesn't have an evaluate function
+   *   has overloaded the evaluate function
+   *   has unsupported param types
+   *   has unsupported return type
+   */
+  @Test def test_UDFClass_noEvaluate() {
+    val schemaFile = Util.daffodilPath("daffodil-udf/src/test/resources/org/apache/daffodil/udf/genericUdfSchema.xsd")
+    val (testSchemaFile) = if (Util.isWindows) (Util.cmdConvert(schemaFile)) else (schemaFile)
```

Comment:
I'm sick of seeing this redundant windows conditionalization, so FYI: I created DAFFODIL-2226 which is to go back to all the CLI-oriented testing not just of the UDFs, but everywhere, and abstract out the need for this conditionalization.